src/plugins/auto/plugin.cpp

```diff
@@ -345,7 +345,10 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
              strDevices += ((iter + 1) == supportDevices.end()) ? "" : ",";
              LOG_INFO("[AUTOPLUGIN]:device:%s, priority:%ld", iter->deviceName.c_str(), iter->devicePriority);
         }
-        return std::make_shared<MultiDeviceExecutableNetwork>(modelPath, network, supportDevices, strDevices, this, context, context.needPerfCounters);
+        // clone the network, in case of reshape conflict
+        CNNNetwork clonedNetwork = InferenceEngine::details::cloneNetwork(network);
```

Comment:
@myshevts what is the `reshape conflict`?

Reply:
this is explained well in the desc of the PR

Reply:
shal the CPU plugin be doing the clone instead?

Reply:
per my observation, CPU plugin also do the clone:
 https://github.com/luo-cheng2021/openvino/blob/75bbfe336d5d2b77bf8d098573214350a095e50e/src/plugins/intel_cpu/src/plugin.cpp#L669
@myshevts @ilya-lavrenov this is more of auto specific issue in **cpu helper** case, because we have gpu loading network and most important, prepare the request queue after the loading**create infer request** in the background threads.  auto will return the exenetwork once cpu done the loading, after which it's open to user to do any other supported operations, like reshape the model. then the conflict is in the previously compiled gpu graphs vs. the new updated model input parameters.

Reply:
I think GPU did the clone as well.
the problem is here:
https://github.com/songbell/openvino/blob/a365ee768b406e41b197809ba7f40a4646890a9e/src/inference/src/cpp_interfaces/interface/ie_iplugin_internal.cpp#L188
GPU is load the network in background thread, after loaded(before which reshape is performed), the _network inputs (old shape) and _parameter (new shape) are mismatch
so to workaround, we can clone the network in auto, but also need to check why the above shapes are not matching, I assume they should be same.

Reply:
sure, our concern is why the cloning helps

Reply:
@myshevts The flow of this issue is:

1. Test app load model to AUTO
2. AUTO load model to CPU for acceleration 
3. AUTO load model to GPU as target device
4. AUTO complete loading model to CPU and create infer request for model on CPU
5. Test app get return for loading model and do model reshape.
6. AUTO complete loading model to GPU and create infer request for model on GPU

Then ExecutableNetwork on GPU in step 3 is based on old shape of model but InferRequest on GPU in step 6 is based on new shape of model. This mismatch cause the exception of this issue.

When AUTO add cloning as this PR, InferRequest on GPU in step 6 will use old shape of model. Then issue is fixed.